File: admin-api/src/mujahid_admin/db/connection.py
from __future__ import annotations

from motor.motor_asyncio import AsyncIOMotorClient, AsyncIOMotorDatabase

from mujahid_admin.core.config import settings

_client: AsyncIOMotorClient | None = None


def get_client() -> AsyncIOMotorClient:
    global _client
    if _client is None:
        _client = AsyncIOMotorClient(settings.mongo_uri)
    return _client


def get_db() -> AsyncIOMotorDatabase:
    return get_client()[settings.mongo_db]


async def ping_mongo() -> bool:
    try:
        await get_client().admin.command("ping")
        return True
    except Exception:
        return False


async def close_mongo() -> None:
    global _client
    if _client is not None:
        _client.close()
        _client = None

File: admin-api/src/db/connection.ts
import { MongoClient, type Db } from "mongodb";
import { settings } from "../core/config";

let client: MongoClient | null = null;

export function getClient(): MongoClient {
  if (!client) {
    client = new MongoClient(settings.mongoUri);
  }
  return client;
}

export function getDb(): Db {
  return getClient().db(settings.mongoDb);
}

export async function pingMongo(): Promise<boolean> {
  try {
    await getClient().db("admin").command({ ping: 1 });
    return true;
  } catch {
    return false;
  }
}

export async function closeMongo(): Promise<void> {
  if (client) {
    await client.close();
    client = null;
  }
}
